packages/calendar/react/src/components/calendar-item/calendar-item.tsx

```diff
@@ -218,23 +233,40 @@ ${extractEventHours(event)}`} // This is not a lint error. The space is left her
     events.forEach((event) => {
       const { id, title, startDate, endDate, account, attendees } = event;
 
-      const backgroundColor = account?.color.color;
-      const textColor = account?.color.darkColor;
+      const eventColor = themeColors[account?.eventColor];
 
-      const isDeclined = attendees?.some(
-        (a) =>
+      const isDeclined = attendees?.some((a) => {
+        return (
           a.email === currentUser?.email &&
-          a.responseStatus === EventResponseStatus.DECLINED,
+          a.responseStatus === EventResponseStatus.DECLINED
+        );
+      });
+
+      // NEEDS_ACTION style
+      const attendeeMe = attendees?.find(
+        (attendee) => attendee.email === account.email,
       );
 
+      const backgroundColor =
+        attendeeMe?.responseStatus !== EventResponseStatus.ACCEPTED
+          ? eventColor[3]
+          : eventColor[7];
+
+      const borderColor =
+        attendeeMe?.responseStatus !== EventResponseStatus.ACCEPTED
+          ? eventColor[7]
+          : eventColor[1];
+
+      const textColor = eventColor[11];
+
```

Comment:
all of this could be done on the server?

Reply:
Yes, that's right. 👍 

Not sure if I want to merge this in before moving that to the server to do it later though.

Will work on directoryFriends's colors first (in attendees.tsx & confirmation-pane.tsx) and then decide to move server side now vs adding a comment to do this later. Probably the latter given that this PR is open for already too long. 😓